client/src/js/services/AccountService.js

```diff
@@ -6,30 +6,64 @@ AccountService.$inject = ['$http', 'util'];
 /**
 * Account Service
 *
-* A service wrapper for the /accounts HTTP endpoint.  
+* A service wrapper for the /accounts HTTP endpoint.
 */
 function AccountService($http, util) {
   var service = this;
-  var baseUrl = '/accounts';
+  var baseUrl = '/accounts/';
 
+  service.read = read;
+  service.label = label;
+
+  /** @deprecated */
   service.list = list;
+
   service.getBalance = getBalance;
   service.getChildren = getChildren;
+
   service.flatten = flatten;
   service.order = order;
 
+  /**
+   * The read() method loads data from the api endpoint. If an id is provided,
+   * the $http promise is resolved with a single JSON object, otherwise an array
+   * of objects should be expected.
+   *
+   * @param {Number} id - the id of the account to fetch (optional).
+   * @param {Object} options - options to be passed as query strings (optional).
+   * @return {Promise} promise - resolves to either a JSON (if id provided) or
+   *   an array of JSONs.
+   */
+  function read(id, options) {
+    var url = baseUrl.concat(id || '');
+    return $http.get(url, { params : options })
+      .then(util.unwrapHttpResponse)
+      .then(function (accounts) {
+
+        // label the accounts with a nice human readable label
+        if (!id) {
```

Comment:
Thanks for your feedback.  I agree that `Angular.isArray` is clearer to read, and probably safer.  An id of 0 shouldn't ever exist in our database...but I'll refactor this for clarity's sake.